test/gtest/ucm/malloc_hook.cc

```diff
@@ -241,8 +249,10 @@ void test_thread::test() {
     /* Release old pointers (should not crash) */
     old_ptrs.clear();
 
+    pthread_mutex_lock(&m_stats_lock);
```

Comment:
Do you mean replace the lock with barrer?
Really I'm not sure this synchronization is really needed. `m_stats_lock` protects only `m_map_ranges` and `m_unmap_ranges` vectors which seems like per thread objects. Adding extra sync decrease number of failures but not fix it. The root cause of failures is a 0-length range in unmap event. I'm going to recheck the fix w/o `m_stats_lock` at all.
It would be helpful if you may confirm 
1) 0-length of range after unmap is a good case
2) thread locality of the vectors